api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -252,5 +258,181 @@ else if (!isObsGroup) {
 			}
 		}
 	}
+
+	/**
+	 * This method validates Obs' numeric values:
+	 * <ol>
+	 *     <li>Validates Obs in relation to criteria e.g. checks patient's age is within the valid range</li>
+	 *     <li>Validates if Obs' numeric value is within the valid range; i.e. >= low absolute && <= high absolute.</li>
+	 *     <li>Sets field errors if numeric value is outside the valid range</li>
+	 * <ol/>
+	 *
+	 * @param obs Observation to validate
+	 * @param errors Errors to record validation issues
+	 *               
+	 * @since 2.7.0
+	 */
+	private void validateConceptReferenceRange(Obs obs, Errors errors) {
+		ConceptReferenceRange conceptReferenceRange = getReferenceRange(obs.getConcept(), obs.getPerson());
+
+		if (conceptReferenceRange != null) {
+			validateAbsoluteRanges(obs, conceptReferenceRange, errors);
+			
+			if (obs.getId() == null) {
+				setObsReferenceRange(obs, conceptReferenceRange);
+			}
+		} else if (obs.getId() == null) {
+			setObsReferenceRange(obs);
+		}
+	}
+
+	/**
+	 * Evaluates the criteria and return the most strict {@link ConceptReferenceRange} for a given concept.
+	 * It considers all valid ranges that match the criteria for the person.
+	 *
+	 * @param concept The concept to evaluate
+	 * @param person The person for whom the range is being evaluated
+	 * @return The strictest {@link ConceptReferenceRange}, or null if no valid range is found
+	 * 
+	 * @since 2.7.0
+	 */
+	public static ConceptReferenceRange getReferenceRange(Concept concept, Person person) {
```

Comment:
Is there any reason why you made these methods static? We try to avoid them as much as we can.

Reply:
Yes, the main reason was because this method is called from this static method `isValidNumericValue`

Reply:
It could instantiate the validator object.